src/main/java/bio/terra/service/dataproject/DataLocationService.java

```diff
@@ -60,6 +61,13 @@ public GoogleProjectResource getProjectForFile(FSFile fsFile) {
     }
 
     public GoogleBucketResource getBucketForFile(FSFile fsFile) {
+        // If this isn't the first time we've seen this file, it should have a bucket resource id we can look up
+        Optional<GoogleBucketResource> bucketForFile = Optional.ofNullable(fsFile.getBucketResourceId())
+            .map(bucketResourceId -> resourceService.getBucketResourceById(UUID.fromString(bucketResourceId)));
```

Comment:
OK, that is an obscure bit of code. I would write it as:
```
if (fsFile.getBucketResourceId() != null) {
return resourceService.getBucketResourceById(UUID.fromString(fsFile.getBucketResourceId());
}
```
I am not scared of testing for null :)